Removing lenses from integration tests (state, addressPoolGap, balance*)

piotr-iohk opened the pull request:
# Issue Number

#1237

# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- b43b949ffa63acf4e7f8a7d88da0d7df92dc1844
  Remove addressPoolGap lens
  
- 24d318ca75b8eccf732f1b0bfbbf01fde4944e92
  Remove state lens
  
- 446f7812916ca96380fb7f4946a4a057e6f7b634
  Remove balance lenses:  - byronBalanceAvailable  - balanceAvailable  - byronBalanceTotal  - balanceTotal  - balanceReward
  
- 9d2e67981eeb63698dc69994c758013601e8802f
  get rid of duplicate magic constants
  - addressPoolGapMin
  - addressPoolGapMax
  - passphraseMinLength
  - passphraseMaxLength

Also got rid of an extra print statement that slipped through

# Comments


<!-- Additional comments or screenshots to attach if any -->

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Assign the PR to a corresponding milestone
 ✓ Acknowledge any changes required to the Wiki
-->

Commit:
Remove addressPoolGap lens

Commit:
Remove state lens

Commit:
Remove balance lenses:  - byronBalanceAvailable  - balanceAvailable  - byronBalanceTotal  - balanceTotal  - balanceReward

KtorZ commented on lib/core-integration/src/Test/Integration/Framework/TestData.hs:
```diff
@@ -288,10 +289,10 @@ passphraseMinLength = 10
 
-addressPoolGapMin :: Int
+addressPoolGapMin :: Word32
 addressPoolGapMin = 10
```
What about using an `AddressPoolGap` type and relying on `minBound` ?

KtorZ replied:
done in 9d2e679

KtorZ commented on lib/core-integration/src/Test/Integration/Framework/TestData.hs:
```diff
@@ -288,10 +289,10 @@ passphraseMinLength = 10
 addressPoolGapMin = 10
 
-addressPoolGapMax :: Int
+addressPoolGapMax :: Word32
```
What about using an `AddressPoolGap` type and relying on `maxBound` ?

KtorZ replied:
done in 9d2e679

KtorZ commented on lib/core-integration/src/Test/Integration/Scenario/API/Addresses.hs:
```diff
@@ -182,7 +182,9 @@ spec = do
-        expectEventually ctx (Link.getWallet @'Shelley) balanceAvailable 10 rb
+        expectEventually ctx (Link.getWallet @'Shelley)
+                (#balance . #getApiT . #available) (Quantity 10) rb
+        print rb
```
An extra `print rb` has sneaked into this code :)

KtorZ committed:
get rid of duplicate magic constants

- addressPoolGapMin
- addressPoolGapMax
- passphraseMinLength
- passphraseMaxLength

Also got rid of an extra print statement that slipped through
